### PR TITLE
review: auto-detect PR base branch for stacked PRs

### PR DESCRIPTION
**Pull request by alexeieleusis**
## Summary
- `vibe-heal review` now auto-detects the current branch's real GitHub PR base via `gh pr view --json baseRefName` instead of always diffing against `origin/main` — fixes duplicate issue reports on stacked PRs (branch B opened against unmerged branch A).
- New `GitHubReviewClient.detect_pr_base_branch()` (never raises — degrades to `None` on any failure) and `ReviewOrchestrator._resolve_base_branch()` (explicit `--base-branch` always wins and skips the `gh` call; omitted falls back to `origin/main` when no PR is found).
- `cleanup` and `dedupe-branch` are intentionally untouched — same `origin/main`-only default as before.
- Docs updated (`docs/review-guide.md`) with a new "Stacked PRs" section; includes a fix for an edge case (non-dict JSON from `gh`) found during a pre-merge security review.

## Test plan
- [x] `uv run pytest -q` — 684 passed
- [x] `uv run mypy` — clean (52 source files)
- [x] `uv run pre-commit run -a` — all hooks pass
- [x] Manually verified `vibe-heal review --help` shows the new auto-detect help text with no visible default, while `vibe-heal cleanup --help` / `vibe-heal dedupe-branch --help` are unchanged (`origin/main` default)

Spec and implementation plan (with full per-task review log): `~/agents/projects/review-base-branch-autodetect/`

### CODE REVIEW

**Review by alexeieleusis — COMMENTED** [review]
SonarQube: no findings on or near changed lines.

File: tests/review/test_github.py [hunk 1]
```diff
@@ -98,6 +98,115 @@ async def _run(cmd, **kwargs):
...
+                stdout='{"baseRefName": "main"}',
+            ),
+        )
+
+        client = GitHubReviewClient()
+        result = await client.detect_pr_base_branch()
+        assert result == "main"
+
+    @pytest.mark.asyncio
+    async def test_returns_none_when_gh_not_installed(self, mocker) -> None:
```
Comment (alexeieleusis) [inline tests/review/test_github.py:150]: `test_returns_none_when_gh_not_installed` (and its docstring, "gh is missing") doesn't actually simulate gh being missing — it mocks `run_command` to *return* `success=False`, which only exercises the `if not result.success: return None` branch (github.py:102-103).

When `gh` is genuinely not installed, `run_command` doesn't return a failed `CommandResult` — `asyncio.create_subprocess_exec` raises `FileNotFoundError` (a subclass of `OSError`), which propagates out of `run_command` uncaught. That's exactly what the `try/except Exception: return None` at github.py:97-100 exists to guard against, per `detect_pr_base_branch`'s own docstring guarantee that it "never raises."

None of the 7 new tests in `TestDetectPrBaseBranch` make `run_command` raise — they all return `MagicMock`s (via `return_value` or `side_effect` functions that return, never raise). So the except-and-swallow path added specifically for the missing-`gh`-binary case is untested.

Suggested fix: add/rename a case that raises, e.g.:
```python
mocker.patch(
    "vibe_heal.review.github.run_command",
    new_callable=AsyncMock,
    side_effect=FileNotFoundError("gh not found"),
)
```
and keep the current `success=False` case under a name like `test_returns_none_when_gh_pr_view_fails` since that's what it actually verifies.

**Comment by alexeieleusis** [comment]
# Review Summary
This PR fixes duplicate issue reports on stacked PRs by having `vibe-heal review` auto-detect the real GitHub PR base branch (via `gh pr view`) instead of always diffing against `origin/main`, while explicit `--base-branch` and the `cleanup`/`dedupe-branch` commands remain unchanged. No blocking issues found. The static-analysis pass on this branch reported 0 issues across the 7 changed files, so no overlapping static analysis findings.

File: tests/review/test_github.py (hunk 1 above)
Comment (alexeieleusis) [reply tests/review/test_github.py:150]: Good catch — the old test mocked `success=False` which only hits the `if not result.success` branch, never the `except Exception` guard. Renamed it to `test_returns_none_when_gh_pr_view_fails` (that's what it actually verifies) and added a new `test_returns_none_when_gh_not_installed` that makes `run_command` raise `FileNotFoundError`, which now exercises the except-and-swallow path.

_https://github.com/alexeieleusis/vibe-heal/pull/169#discussion_r3574007634_
_https://github.com/alexeieleusis/vibe-heal/commit/2a67e841c7d0d1b6cf03e38346241858e624c9d6_
